presto-main/src/main/java/com/facebook/presto/operator/ChannelSet.java

```diff
@@ -29,12 +30,14 @@
 {
     private final GroupByHash hash;
     private final boolean containsNull;
+    private final long elements;
```

Comment:
you can just have `isEmpty` flag. `elements` suggest that this number is somehow reliable (as per total number of elements in cluster). What you are interested is only if the set is empty (on whole cluster).